### PR TITLE
New read me

### PR DESCRIPTION
**Pull request by nicola-giuliani**

### CODE REVIEW

**Comment by nicola-giuliani** [comment]
Updated the README providing some details on the library.

File: README.md [hunk 1]
```diff
@@ -1,8 +1,61 @@
-Parallel BEM Solver for deal.II
-===============================
+#Parallel BEM Solver for deal.II
 
 [![Build Status](https://travis-ci.org/mathLab/pi-BEM.svg?branch=master)](https://travis-ci.org/mathLab/pi-BEM)
 
+The library represents a parallel solver for the Laplace equation through Boundary Element Methods. We have developed the software in C++ on top of the [deal.II](https://github.com/dealii/dealii) library.
```
Comment (asartori86) [inline README.md]: represents --> features, provides?

File: README.md [hunk 2]
```diff
@@ -1,8 +1,61 @@
...
 
 [![Build Status](https://travis-ci.org/mathLab/pi-BEM.svg?branch=master)](https://travis-ci.org/mathLab/pi-BEM)
 
+The library represents a parallel solver for the Laplace equation through Boundary Element Methods. We have developed the software in C++ on top of the [deal.II](https://github.com/dealii/dealii) library. 
+
+## Provided features
+
+We provide the following capabilities
+
+- Read of the grid through an external file in one of the following formats (.prm, .msh, .vtk), the files specifies the kind of boundary condition to be applied on the nodes.
```
Comment (asartori86) [inline README.md]: read of --> input

File: README.md [hunk 3]
```diff
@@ -1,8 +1,61 @@
...
+- Possibility of solving mixed Dirichlet Neumann boundary value problem.
+- Automatic treatment of sharp edges via the double nodes technique.
+- Usage of Lagrangian Finite Elements of arbitrary order. We also provide interfaces with discontinuous elements.
+- Distributed memory (MPI) parallelisation of the standard collocation BEM for the Laplace equation.
+- Coupling with a FMM to get a performance improvement.
```
Comment (asartori86) [inline README.md]: FMM = ? :)

File: README.md [hunk 4]
```diff
@@ -1,8 +1,61 @@
...
+- Hybrid Distributed (MPI) - Shared (TBB) memory parallelisation for the BEM-FMM code
+- Recovery of both primal (potential) and dual (potential normal derivative) unknowns.
+- L2 projection  of the full 3D potential gradient for post processing.
```
Comment (asartori86) [inline README.md]: doppio spazio :)

File: README.md [hunk 5]
```diff
@@ -1,8 +1,61 @@
...
+- Extensive tuning via parameter file using the [deal2lkit](https://github.com/mathLab/deal2lkit) library 
+
+
+## Code Structure
+We have subdivided the code in main classes to handle the many different aspects of a complete BEM simulation.
+
+- Driver. This class is in charge of organising the overall BEM simulation. It has interfaces with all the other classes in order to have a complete simulation.
```
Comment (asartori86) [inline README.md]: in order to have or in order to perform?

File: README.md [hunk 6]
```diff
@@ -1,8 +1,61 @@
...
+- ComputationalDomain. This class handles, and provides to the other classes, ONLY the geometry of the problem. In particular
+ - it handles the domain decomposition using a graph partitioning tool (METIS);
+ - it reads the domain from an external file.
+- BoundaryCondition. The class handling the boundary conditions. In particular
```
Comment (asartori86) [inline README.md]: handles

File: README.md [hunk 7]
```diff
@@ -1,8 +1,61 @@
...
+	- it reads the boundary conditions for the potential and its normal derivative; 
+	- given the peculiarities of the BEM the boundary conditions represent the actual unknowns, thus it creates the vector containing the variables and fills them with the proper data;
```
Comment (asartori86) [inline README.md]: add a `,` after BEM

Comment (asartori86) [inline README.md]: I guess you fill the *vector* not the *variables*. If so them--> it

File: README.md [hunk 8]
```diff
@@ -1,8 +1,61 @@
...
+	- it performs the error analysis on both unknowns.
+
+- BEMProblem. This class is the core of the BEM simulation
+	- it receives the variables vector filled in with the proper boundary condition;
+	- it creates the codimension 1 functional space setting up the BEM;
+	- it solves the system using a preconditioned parallel GMRES solver;
+	- it eventually interacts with the FMM accelerator.
+- BEMFMA. This class handles the accelerator, in particular
+	- it sets up an hierarchical 3D space subdivision (octree);
+	- it receives two distributed vectors representinf the unknowns and performs a full FMM matrix vector product.
```
Comment (asartori86) [inline README.md]: representing

File: README.md [hunk 9]
```diff
@@ -1,8 +1,61 @@
...
+
+## Install Procedure
+In order to successfully compile the code you need 
+
+- to install the [deal.II](https://github.com/dealii/dealii) library allowing both for multiprocessors and multithreaded environment.
+- to install the Trilinos and Metis wrappers of the library, see the official [instructions](https://www.dealii.org/developer/readme.html)
```
Comment (asartori86) [inline README.md]: I guess it is better to install Trilinos first and dealii afterwards :).  deal2lkit?

File: include/boundary_conditions.h [hunk 10]
```diff
@@ -76,6 +76,12 @@
 #include <deal2lkit/parsed_data_out.h>
 #include <deal2lkit/utilities.h>
 
+/**
+* - BoundaryCondition. The class handling the boundary conditions. In particular
+*   - it reads the boundary conditions for the potential and its normal derivative;
+*   - given the peculiarities of the BEM the boundary conditions represent the actual unknowns, thus it creates the vector containing the variables and fills them with the proper data;
```
Comment (asartori86) [inline include/boundary_conditions.h]: same comment as above

**Comment by nicola-giuliani** [comment]
Thank you Alberto for all the comments, added and pushed. ;)

**Pull request merged**